### PR TITLE
Support Date32 data type.

### PR DESCRIPTION
**Pull request by huyphams**
- Added date32 (Which is 4 bytes when compared with 2 byte Date).

### CODE REVIEW

**Comment by huyphams** [comment]
I think I should add a test for Date32

**Comment by huyphams** [comment]
I think I should add a test for Date32

**Comment by Enmk** [comment]
> I think I should add a test for Date32

Hi, thank you for contribution!

Yes, please add some tests. There are two kinds of them:
- unit-tests (`ut/columns_ut.cpp`), where one should test APIs of Column: creating, appending, getting values, slicing, etc.
- integration-tests (`ut/client_ut.cpp`), where one should test roundtrip of Column data to/from CH server.

If you find it difficult adding both, please add at least something and allow `edit from maintainers` in this PR.

File: clickhouse/types/types.h [hunk 1]
```diff
@@ -34,6 +34,7 @@ class Type {
         FixedString,
         DateTime,
         Date,
+        Date32,
```
Comment (Enmk) [inline clickhouse/types/types.h:37]: Please move the new value to the end. This is `enum` is visible to users and some may rely on order of elements. I think we can re-arrange those to look better in a next major version, like `3.0`

**Review by Enmk — CHANGES_REQUESTED** [review]
Please fix minor issue mentioned

**Comment by huyphams** [comment]
Thank you for reviewing, I will fix it 🙌

File: clickhouse/types/types.h (hunk 1 above)
Comment (huyphams) [reply clickhouse/types/types.h]: Done

**Comment by Enmk** [comment]
Please rebase/merge master and fix the builds

**Comment by huyphams** [comment]
Hi @Enmk I've merged master into this branch. Could you help me to write the test (if you have time)? 🙌

**Comment by Enmk** [comment]
> Hi @Enmk I've merged master into this branch. Could you help me to write the test (if you have time)? raised_hands

Sure, I've drafted a scaffold for unit-testing Column implementation. Will try to finalize and commit it on Monday.

**Comment by Enmk** [comment]
> Hi @Enmk I've merged master into this branch. Could you help me to write the test (if you have time)? raised_hands

Hello @huyphams !
It took a bit more time to iron everything out, but it is almost ready now:
https://github.com/ClickHouse/clickhouse-cpp/pull/179

In order to benefit form test suite, you'll need to make some changes to the `ut/Column_ut.cpp`:
- add ColumnDateTime32 to the type-list:
```using ValueColumns = ::testing::Types<```
- add meaningful value generator to 
```static auto GenerateValues(size_t values_size) {```
- optionally add proper constructor to static auto MakeColumn() {
```static auto MakeColumn() {```


Also you are welcome to review the pr #179

**Comment by Enmk** [comment]
@huyphams ping?

**Pull request merged**